src/phoenix/server/api/routers/utils.py

```diff
@@ -19,3 +21,73 @@ def from_iso_format(value: Optional[str]) -> Optional[datetime]:
 def df_to_bytes(df: pd.DataFrame) -> bytes:
     pa_table = pa.Table.from_pandas(df)
     return table_to_bytes(pa_table)
+
+
+def prepend_root_path(*, request: Request, path: str) -> str:
+    """
+    Prepends the ASGI root path to the given path if one is configured.
+
+    Automatically normalizes paths by ensuring leading slashes are present
+    and removing trailing slashes to prevent double slashes in the result.
+
+    Args:
+        request: The FastAPI/Starlette request object containing ASGI scope
+        path: The path to prepend the root path to (e.g., "/login", "logout")
+
+    Returns:
+        The normalized path with root path prepended if configured,
+        otherwise just the normalized path.
+
+    Examples:
+        With root_path="/api/v1":
+        - prepend_root_path_if_exists(request, "/login") -> "/api/v1/login"
+        - prepend_root_path_if_exists(request, "login") -> "/api/v1/login"
+
+        With no root_path:
+        - prepend_root_path_if_exists(request, "/login") -> "/login"
+        - prepend_root_path_if_exists(request, "login") -> "/login"
```

Comment:
this is a bit of a confusing example with `/api/v1` - better to make a realistic example
```suggestion
    Examples:
        With root_path="/apps/phoenix":
        - prepend_root_path_if_exists(request, "/login") -> "/apps/phoenix/login"
        - prepend_root_path_if_exists(request, "login") -> "/apps/phoenix/login"

        With no root_path:
        - prepend_root_path_if_exists(request, "/login") -> "/login"
        - prepend_root_path_if_exists(request, "login") -> "/login"
```